main.rb

```diff
@@ -0,0 +1,94 @@
+module Enumerable
+  def my_each
+    index = 0
+
+    while index < size
+      yield self[index]
+      index += 1
+    end
+  end
+
+  def my_each_with_index
+    item = 0
+    index = 0
+
+    while index < size
+      yield self[item], index
+      index += 1
+      item += 1
+    end
+  end
+
+  def my_select
+    arr = []
+
+    my_each do |item|
+      arr.push(item) if yield item
+    end
+    arr
+  end
```

Comment:
- [ ] should have working `my_select` **(covered by automated tests)** identical and returns the same thing as ruby's [`select`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-select). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_select.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

Yours is not Returning an enumerator when no block is given.